### PR TITLE
[#5023] Dynamically set window window and height

### LINKED ISSUE #5023: [ipad] `Fetching messages` or `Offline` grey line is shorter than screen length in landscape mode
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: `Fetching messages` or `Offline` grey line in desktop mode is shorter than screen length in landscape mode
img-0429

### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)
on 00:56 https://app.testfairy.com/projects/4803590-status/builds/8509899/sessions/4395293660/?accessToken=11TMA-oAT4KQQxZONXGAmy8u-Qo
- Open Status and login in portrait mode on iPad
- Turn iPad to landscape mode 
- Join any chat to get "Fetching messages" grey line shown and check its length

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: PR build https://github.com/status-im/status-react/pull/4720#issuecomment-401107276
* Operating System:  iPad mini 3, iOS 11.4

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803590-status/builds/8509899/sessions/4395293660/?accessToken=11TMA-oAT4KQQxZONXGAmy8u-Qo">TestFairy: Status 0.9.21 (6133) @2018</a></strong></div></blockquote>

### PR DESCRIPTION
**Pull request by dmitryn**
fixes #5023

### Summary:

Listen to window dimensions change event to dynamically set window width and height.

https://facebook.github.io/react-native/docs/dimensions.html

### Testing notes (optional):
Please test app in horizontal/landscape layout on tablets.

status: ready

<blockquote><div><strong><a href="https://facebook.github.io/react-native/index.html">Dimensions · React Native</a></strong></div><div>### Methods</div></blockquote>

### CODE REVIEW

**Review by goranjovic — APPROVED** [review]

**Comment by goranjovic** [comment]
@annadanchenko pushed the fix for both issues

File: src/status_im/ui/components/qr_code_viewer/views.cljs [hunk 1]
```diff
@@ -18,7 +19,7 @@
 
 (defn qr-code-viewer [{:keys [style hint-style footer-style]} value hint legend]
   {:pre [(not (nil? value))]}
-  (let [{:keys [width height]} (react/get-dimensions "window")]
+  (let [{:keys [width height]} (re-frame/subscribe [:dimensions/window])]
```
Comment (jeluard) [inline src/status_im/ui/components/qr_code_viewer/views.cljs]: Could be using defview?

**Review by jeluard — APPROVED** [review]

File: src/status_im/ui/components/qr_code_viewer/views.cljs (hunk 1 above)
Comment (dmitryn) [reply src/status_im/ui/components/qr_code_viewer/views.cljs]: @jeluard i thought about that too, but not sure `:pre` (or passing fn metadata in general) is supported there.

Comment (jeluard) [reply src/status_im/ui/components/qr_code_viewer/views.cljs]: Good point

**Comment by yevh-berdnyk** [comment]
Please ignore tests failed with `Password in logcat!!!`, it's false failure and will be fixed soon

**Comment by annadanchenko** [comment]
### issue 1
QR codes are not shown on 2 screens
1. Wallet -> Receive Transaction screen
img_2682

2. Profile -> Share my contact code 
img_2681

it happens on iOS and Android real devices (iPad mini 3, iPhone 6s and Pixel XL)
Note that in the nightly build with ipad PR merged both codes are shown fine, so I assumed it's PR's issue.

**Comment by dmitryn** [comment]
@annadanchenko i've fixed the issues, please take a look.

**Pull request merged**